client/client.go

```diff
@@ -59,12 +61,16 @@ const (
 // To create a new client, please provide the url of the flyte server and the timeout.
 // timeout specifies a time limit for requests made by this
 // client. A timeout of zero means no timeout.
-func NewClient(rootURL *url.URL, timeout time.Duration) Client {
+func NewClient(rootURL *url.URL, timeout time.Duration, IsInsecure bool) Client {
 	baseUrl := getBaseURL(*rootURL)
+
 	client := &client{
 		baseURL: baseUrl,
 		httpClient: &http.Client{
 			Timeout: timeout,
+			Transport:  &http.Transport{
```

Comment:
The docs for this state "	// This should be used only for testing." - why are we adding it?

Reply:
They may say that but it's used alot not just for testing